ls1intum/Artemis
# `General`: Fix issues with tutor performance in assessment dashboard

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Instructor is interested in cases when a particular tutor performs worse than the average of the group. It would be nice to see which tutors have issues with performance and what exactly is wrong: bad scoring, bad rating or a lot of complaints? This PR solves this by computing the issues with tutor performance and showing it as a list in course assessment dashboard as well as marking the tutors with bad performance in the tutor leaderboard with ⚠️ near their name.

### Description

Issues with tutor performance are computed on the fly the client side. `TutorIssueRatingChecker`, `TutorIssueScoreChecker`, `TutorIssueComplaintsChecker` check if tutor has a performance issue with rating, score or complaints.
When it comes to complaints, complaints per submission ratio is computed, this way the performance issue is computed fairly.

Tutor performance issue is computed by computing the course average +/- 20%. This is the threshold value. If the tutor surpasses it, then he/she has a performance issue, which will be shown in the list.

### Steps for Testing

1. Log in to Artemis as Instructor
2. Navigate to Course Administration
3. Navigate to Course Assessment Dashboard
4. You should see the list with issues with tutor performance

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






#### Before
<img width="1840" alt="before" src="https://user-images.githubusercontent.com/51077603/138154662-06284d8e-bb38-4438-b07f-e7bfba92ef74.png">

#### After - logged in as Instructor
<img width="1223" alt="Screenshot 2021-10-22 at 3 49 47 PM" src="https://user-images.githubusercontent.com/51077603/138465343-ac415bbb-7b6d-4c2f-8cc6-3dfbb73e7907.png">
